flatgeobuf/hilbert-r-tree.qmd

```diff
@@ -0,0 +1,52 @@
+---
+title: FlatGeobuf Spatial Index
+subtitle: Breaking down the Hilbert Packed R-Tree
+---
+
+FlatGeobuf supports optionally including a spatial index that enables random access for each geometry in the file.
+
+## When to use
+
+When _writing_ a FlatGeobuf file, one must decide whether to include a spatial index. A spatial index cannot be added to a FlatGeobuf file after the file has been written.
+
+A spatial index can enable much more efficient reading from FlatGeobuf, by allowing the reader to skip over portions of the file that fall outside of a qiven spatial query region.
+
+## Technical details
+
+In this section we'll get into some gory technical details of how FlatGeobuf's spatial index works. Understanding the below isn't necessary for using FlatGeobuf, but can provide some extra context.
+
+FlatGeobuf's spatial index is a [_static packed Hilbert R-tree index_](https://flatgeobuf.org/#specification). That's a mouthful, so let's break it down:
+
+### R-tree index
+
+An R-Tree is a hierarchical collection of bounding boxes. From the bottom up, at the lowest level of the tree is a bounding box of every geometry. Then one level above the bottom holds a collection of bounding boxes, each of which is formed as the union of all child boxes. This means that each box encompasses every child box. There are fewer boxes at this level, because each box contains many child boxes, each of which represents one original geometry. This process continues repeatedly until there's only one bounding box that indirectly contains the entire dataset.
```

Comment:
```suggestion
An R-Tree is a hierarchical collection of bounding boxes. At the lowest level of the tree is a bounding box of every geometry. Then one level above the lowest level exists a collection of bounding boxes, each of which is formed as the union of all child boxes. This means that each box encompasses every child box. There are fewer boxes at this level, because each box contains many child boxes, each of which represents one original geometry. This process continues repeatedly until there's only one bounding box that indirectly contains the entire dataset.
```